enterprise/suppress-user/internal/badgerdb/badgerdb_test.go

```diff
@@ -28,125 +23,160 @@ func (f readerFunc) Read(p []byte) (n int, err error) {
+	// 	}), badgerdb.WithMaxSeedWait(1*time.Millisecond))
+	// 	require.NoError(t, err)
+	// })
+
+	// t.Run("try to restore invalid data", func(t *testing.T) {
+	// 	r := bytes.NewBuffer([]byte("invalid data"))
+	// 	require.Error(t, repo.Restore(r), "it should return an error when trying to restore invalid data")
+	// })
+
+	// t.Run("badgerdb errors", func(t *testing.T) {
+	// 	require.NoError(t, repo.Stop(), "it should be able to stop the badgerdb instance without an error")
+
+	// 	_, err := repo.Suppressed("workspace1", "user1", "")
+	// 	require.Error(t, err)
+
+	// 	_, err = repo.GetToken()
+	// 	require.Error(t, err)
+
+	// 	require.Error(t, repo.Add([]model.Suppression{}, []byte("")))
+
+	// 	require.Error(t, repo.Add([]model.Suppression{{
+	// 		WorkspaceID: "workspace1",
+	// 		UserID:      "user1",
+	// 		SourceIDs:   []string{},
+	// 	}}, []byte("token")))
+	// })
+
+	// t.Run("trying to use a closed repository", func(t *testing.T) {
+	// 	repo, err := badgerdb.NewRepository(basePath, logger.NOP)
+	// 	require.NoError(t, err)
+	// 	require.NoError(t, repo.Stop())
+
+	// 	require.Equal(t, repo.Add(nil, nil), badger.ErrDBClosed)
+
+	// 	s, err := repo.Suppressed("", "", "")
+	// 	require.False(t, s)
+	// 	require.Equal(t, err, badger.ErrDBClosed)
+
+	// 	_, err = repo.GetToken()
+	// 	require.Equal(t, err, badger.ErrDBClosed)
+
+	// 	require.Equal(t, repo.Backup(nil), badger.ErrDBClosed)
+
+	// 	require.Equal(t, repo.Restore(nil), badger.ErrDBClosed)
+	// })
 
 	t.Run("trying to use a closed repository", func(t *testing.T) {
 		repo, err := badgerdb.NewRepository(basePath, logger.NOP)
 		require.NoError(t, err)
-		require.NoError(t, repo.Stop())
-
-		require.Equal(t, repo.Add(nil, nil), badger.ErrDBClosed)
 
-		s, err := repo.Suppressed("", "", "")
-		require.False(t, s)
-		require.Equal(t, err, badger.ErrDBClosed)
+		repo.Add([]model.Suppression{{
+			WorkspaceID: "workspace1",
+			UserID:      "user1",
+			SourceIDs:   []string{"src-1"},
+		}}, []byte("token"))
 
-		_, err = repo.GetToken()
-		require.Equal(t, err, badger.ErrDBClosed)
+		con := func() bool {
+			suppressed, err := repo.Suppressed("workspace1", "user1", "src-1")
+			require.NoError(t, err)
+			return suppressed
+		}
+		require.Eventually(t, con, 5*time.Second, 100*time.Millisecond, "abc")
 
-		require.Equal(t, repo.Backup(nil), badger.ErrDBClosed)
+		tmp := []byte{}
+		err = repo.Backup(bytes.NewBuffer(tmp))
+		require.NoError(t, err)
+		repo2, err := badgerdb.NewRepository(path.Join(t.TempDir(), strings.ReplaceAll(uuid.New().String(), "-", "")), logger.NOP)
+		require.NoError(t, err)
+		err = repo2.Restore(bytes.NewBuffer(tmp))
+		require.NoError(t, err)
 
-		require.Equal(t, repo.Restore(nil), badger.ErrDBClosed)
+		con2 := func() bool {
+			suppressed, err := repo2.Suppressed("workspace1", "user1", "src-1")
+			require.NoError(t, err)
+			return suppressed
+		}
+		require.Eventually(t, con2, 5*time.Second, 100*time.Millisecond, "abc")
 	})
+
```

Comment:
this test seems to be failing at line 179, which is not expected since we are restoring, what we backed up before.